src/script/tari_script.rs

```diff
@@ -138,41 +196,195 @@ impl TariScript {
+            IfThen => self.handle_if_then(stack),
+            Else => self.handle_else(),
+            EndIf => self.handle_end_if(),
+        }
+    }
+
+    fn handle_check_height_verify(height: u64, block_height: u64) -> Result<(), ScriptError> {
+        if block_height >= height {
+            Ok(())
+        } else {
+            Err(ScriptError::VerifyFailed)
+        }
+    }
+
+    fn handle_check_height(stack: &mut ExecutionStack, height: u64, block_height: u64) -> Result<(), ScriptError> {
+        let height = i64::try_from(height)?;
+        let block_height = i64::try_from(block_height)?;
+        let item = StackItem::Number(block_height - height);
+
+        stack.push(item)
+    }
+
+    fn handle_compare_height_verify(stack: &mut ExecutionStack, block_height: u64) -> Result<(), ScriptError> {
+        let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
+        let target_height = if let StackItem::Number(n) = item {
+            Ok(n)
+        } else {
+            Err(ScriptError::InvalidInput)
+        }?;
+        let target_height = u64::try_from(target_height)?;
+
+        if block_height >= target_height {
+            Ok(())
+        } else {
+            Err(ScriptError::VerifyFailed)
+        }
+    }
+
+    fn handle_compare_height(stack: &mut ExecutionStack, block_height: u64) -> Result<(), ScriptError> {
+        let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
+        let target_height = match item {
+            StackItem::Number(n) => n,
+            _ => {
+                return Err(ScriptError::InvalidInput);
+            },
+        };
+        let block_height = i64::try_from(block_height)?;
+
+        let item = StackItem::Number(block_height - target_height);
+
+        stack.push(item)
+    }
+
+    fn handle_cmp_to_zero(stack: &mut ExecutionStack, valid_orderings: &[Ordering]) -> Result<(), ScriptError> {
+        let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
+        let stack_number = if let StackItem::Number(n) = item {
+            Ok(n)
+        } else {
+            Err(ScriptError::InvalidInput)
+        }?;
+
+        let ordering = &stack_number.cmp(&0);
+        if valid_orderings.contains(ordering) {
+            stack.push(StackItem::Number(1))
+        } else {
+            stack.push(StackItem::Number(0))
+        }
+    }
+
+    fn handle_or(stack: &mut ExecutionStack, n: u8) -> Result<(), ScriptError> {
+        let items = stack.pop_num_items(n)?;
+        let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
+
+        if items.contains(&item) {
+            stack.push(StackItem::Number(1))
+        } else {
+            stack.push(StackItem::Number(0))
+        }
+    }
+
+    fn handle_or_verify(stack: &mut ExecutionStack, n: u8) -> Result<(), ScriptError> {
+        let items = stack.pop_num_items(n)?;
+        let item = stack.pop().ok_or(ScriptError::StackUnderflow)?;
+
+        if items.contains(&item) {
+            Ok(())
```

Comment:
This is close, but still probably better to DRY up with the non_verify version. 

Ditto for all xxx_verify